wasm_codec/derive/src/lib.rs

```diff
@@ -382,8 +382,8 @@ fn classify_fn(fn_sig: &syn::Signature) -> FnClass {
 
     let err_type_path = unwrap_path(err_type);
     let err_type_last_segment = last_segment(err_type_path);
-    let err_type = if err_type_last_segment.ident == "Trap" {
-        ErrType::Trap
+    let err_type = if err_type_last_segment.ident == "WasmtimeError" {
```

Comment:
> "WasmtimeError"

This approach is very sensitive to errors - it would be nice if I rewrite it and improve it.

Reply:
Yeah, but I don't know what we can do here

Reply:
Other than introducing a circular dependency, I doubt it's possible.